ssh-key/src/private/sk.rs

```diff
@@ -26,6 +26,17 @@ pub struct SkEcdsaSha2NistP256 {
 
 #[cfg(feature = "ecdsa")]
 impl SkEcdsaSha2NistP256 {
+    /// Construct new instance of SkEcdsaSha2NistP256.
+    #[cfg(feature = "alloc")]
+    pub fn new(public: public::SkEcdsaSha2NistP256, flags: u8, key_handle: &[u8]) -> Self {
+        SkEcdsaSha2NistP256 {
+            public,
+            flags,
+            key_handle: key_handle.to_vec(),
+            reserved: Vec::<u8>::new(),
+        }
+    }
```

Comment:
If this is all the constructor is doing, we could just make the fields public.

I think I made them private initially so these parameters could potentially be validated in some way.

Also it'd be good to use `key_handle: impl Into<Vec<u8>>` to avoid copying if a `Vec` is already available

Reply:
Well... key_handle must be 0-255 B long (https://fidoalliance.org/specs/fido-u2f-v1.1-id-20160915/fido-u2f-raw-message-formats-v1.1-id-20160915.html#h3_registration-response-message-success).

For flags there are currently only two possible values: 0 and 1 (https://fidoalliance.org/specs/fido-u2f-v1.1-id-20160915/fido-u2f-raw-message-formats-v1.1-id-20160915.html#fig-user-presence-byte-layout).

The reserved field should be empty.

Reply:
In that case, it'd be good to make the constructor fallible

Reply:
Better? I don't think that the flag should be checked.

Reply:
Sure. As long as it's fallible we can change what's allowed in the future.